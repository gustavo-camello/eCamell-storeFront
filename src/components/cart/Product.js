import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faMinus, faPlus, faTrash } from "@fortawesome/free-solid-svg-icons";

function Product() {
  return (
    <li className="flex items-center py-4">
      <a href="" className="flex-shrink-0 block">
        <img
          src="https://images.unsplash.com/photo-1600080972464-8e5f35f63d08?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=668&q=80"
          alt=""
          className="object-cover w-24 h-24 rounded-lg"
        />
      </a>

      <div className="w-full ml-8">
        <div className="sm:flex sm:justify-end">
          <div>
            <a href="" className="block font-medium">
              Amazing Controller
            </a>

            <p className="mt-1 text-sm text-gray-500">
              White Edition &middot; $49.99
            </p>
          </div>

          <div className="flex items-center flex-1 mt-4 sm:justify-end sm:mt-0">
            <div className="flex space-x-1">
              <button
                type="button"
                className="flex-shrink-0 px-2 text-gray-500 border border-white hover:border-gray-200 rounded-lg"
              >
                <FontAwesomeIcon icon={faMinus} />
              </button>

              <input
                type="number"
                value="1"
                min="0"
                className="w-12 px-2 text-center text-gray-500 border-none no-spinners"
              />

              <button
                type="button"
                className="flex-shrink-0 px-2 text-gray-500 border border-white hover:border-gray-200 rounded-lg"
              >
                <FontAwesomeIcon icon={faPlus} />
              </button>
            </div>

            <a
              href=""
              className="flex-shrink-0 px-2 py-1 ml-8 text-gray-600 hover:bg-gray-100 rounded-lg"
            >
              <FontAwesomeIcon icon={faTrash} />
            </a>
          </div>
        </div>
      </div>
    </li>
  );
}

export default Product;
