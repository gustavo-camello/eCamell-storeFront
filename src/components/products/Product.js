import React from "react";

import Link from "next/link";
import DisplayMoney from "components/DisplayMoney";

function Product({ product }) {
  return (
    <div className="lg:w-1/4 md:w-1/2 p-4 w-full dark:text-white">
      <Link href={`/product/${product.id}`}>
        <div className="cursor-pointer">
          <div className="block relative h-48 rounded overflow-hidden">
            <img
              alt="ecommerce"
              className="object-cover object-center w-full h-full block"
              src={product?.photo?.image?.publicUrlTransformed}
            />
          </div>
          <div className="mt-4">
            <h3 className="text-xs tracking-widest mb-1 dark:text-gray-400">
              {product.category[0].name}
            </h3>
            <h2 className="text-base capitalize dark:text-gray-200">
              {product.name}
            </h2>
            <p className="mt-2 text-xl font-bold text-yellow-600">
              <DisplayMoney amount={product?.price} />
            </p>
          </div>
        </div>
      </Link>
    </div>
  );
}

export default Product;
