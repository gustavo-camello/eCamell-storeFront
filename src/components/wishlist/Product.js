import React from "react";
import Link from "next/link";

import DisplayMoney from "components/DisplayMoney";
import AddToCart from "components/product/AddToCart";

function Product({ productDetails }) {
  const { name, price, id } = productDetails;
  const showButtons = true;

  return (
    <Link href={`/product/${id}`} className="cursor-pointer">
      <li className="flex flex-col sm:flex-row items-center py-4 cursor-pointer dark:text-white">
        <a className="flex-shrink-0 block">
          <img
            src={productDetails?.photo?.image?.publicUrlTransformed}
            alt="Product Image"
            className="object-cover w-48 h-48 md:w-24 md:h-24 rounded-lg mb-4 md:mb-0"
          />
        </a>

        <div className="sm:w-full sm:ml-8">
          <div className="sm:flex sm:justify-end">
            <div className=" md:w-2/3">
              <span className="block font-light text-left hover:font-semibold">
                {name}
              </span>
              <div className="mt-1 text-sm font-medium flex justify-start items-center text-yellow-600">
                <span>
                  {" "}
                  <DisplayMoney amount={price} />
                </span>
              </div>
            </div>

            <div className="flex w-full items-center justify-center flex-1 mt-4 sm:justify-end sm:mt-0">
              <div className="flex space-x-1 justify-center items-center">
                {showButtons && <AddToCart productId={id} />}
              </div>

              <div>
                {showButtons && (
                  <button
                    type="button"
                    className="flex-shrink-0 px-2 py-1 ml-8 text-gray-600 hover:border-gray-300 hover:bg-gray-200 rounded-lg"
                  ></button>
                )}
              </div>
            </div>
          </div>
        </div>
      </li>
    </Link>
  );
}

export default Product;
