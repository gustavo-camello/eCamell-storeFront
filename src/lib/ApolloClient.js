import { ApolloClient, InMemoryCache } from "@apollo/client";

export const productsClient = new ApolloClient({
  uri: "http://localhost:3000/api/graphql",
  cache: new InMemoryCache(),
});

export const customersClient = new ApolloClient({
  uri: "http://localhost:3010/api/graphql",
  cache: new InMemoryCache(),
});

export const cartClient = new ApolloClient({
  uri: "http://localhost:3020/api/graphql",
  cache: new InMemoryCache(),
});
