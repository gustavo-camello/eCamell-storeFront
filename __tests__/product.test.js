import { render, screen } from "@testing-library/react";
import { MockedProvider } from "@apollo/react-testing";
import "@testing-library/jest-dom";

import Product from "../src/components/products/Product";
import { fakeItem } from "../src/lib/testUtils";

const product = fakeItem();

describe("<Product/>", () => {
  it("renders out the price tag and title", () => {
    const { container, debug } = render(
      <MockedProvider>
        <Product product={product} />
      </MockedProvider>
    );

    expect(screen.getByText("5000")).toBeInTheDocument();

    const link = container.querySelector("a");
    debug(link);
    expect(container).toHaveTextContent(product?.name);
  });

  it("renders and match the snapshot", () => {
    const { container, debug } = render(
      <MockedProvider>
        <Product product={product} />
      </MockedProvider>
    );

    expect(container).toMatchSnapshot();
  });

  it("renders the image properly", () => {
    const { container, debug } = render(
      <MockedProvider>
        <Product product={product} />
      </MockedProvider>
    );

    const img = screen.getByAltText("ecommerce");
    expect(img).toBeInTheDocument();
  });
});
