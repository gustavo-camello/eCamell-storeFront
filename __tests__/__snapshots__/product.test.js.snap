// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product/> renders and match the snapshot 1`] = `
<div>
  <div
    class="lg:w-1/4 md:w-1/2 p-4 w-full dark:text-white"
  >
    <div
      class="cursor-pointer"
    >
      <div
        class="block relative h-48 rounded overflow-hidden"
      >
        <img
          alt="ecommerce"
          class="object-cover object-center w-full h-full block"
          src="dog.jpg"
        />
      </div>
      <div
        class="mt-4"
      >
        <h3
          class="text-xs tracking-widest mb-1 dark:text-gray-400"
        />
        <h2
          class="text-base capitalize dark:text-gray-200"
        >
          dogs are best
        </h2>
        <p
          class="mt-2 text-xl font-bold text-yellow-600"
        >
          5000
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Product/> renders out the price tag and title 1`] = `
<div>
  <div
    class="lg:w-1/4 md:w-1/2 p-4 w-full dark:text-white"
  >
    <div
      class="cursor-pointer"
    >
      <div
        class="block relative h-48 rounded overflow-hidden"
      >
        <img
          alt="ecommerce"
          class="object-cover object-center w-full h-full block"
          src="dog.jpg"
        />
      </div>
      <div
        class="mt-4"
      >
        <h3
          class="text-xs tracking-widest mb-1 dark:text-gray-400"
        />
        <h2
          class="text-base capitalize dark:text-gray-200"
        >
          dogs are best
        </h2>
        <p
          class="mt-2 text-xl font-bold text-yellow-600"
        >
          5000
        </p>
      </div>
    </div>
  </div>
</div>
`;
